clippy_lints/src/methods/mod.rs

```diff
@@ -4094,14 +4094,19 @@ fn lint_from_iter(cx: &LateContext<'_>, expr: &hir::Expr<'_>, args: &[hir::Expr<
         if implements_trait(cx, ty, from_iter_id, &[]) && implements_trait(cx, arg_ty, iter_id, &[]);
         then {
             // `expr` implements `FromIterator` trait
-            let iter_expr = snippet(cx, args[0].span, "..");
+            let iter_expr = sugg::Sugg::hir(cx, &args[0], "..").maybe_par();
+            let sugg = if higher::range(&args[0]).is_some() {
```

Comment:
Is this `if` needed? I don't know whether or not it should be done in here, but the suggested codes in other test cases seem not to work as well, for example `HashMap::<usize, &i8>::from_iter`. Doesn't it work if we remove this?

Reply:
Hmm good catch, I didn't notice this suggestion was wrong also for `HashMap`, so yes maybe we need the turbofish each time.
I will also try to add a rustfix here, as I want to be sure the suggestions work and compile fine.